vumi_message_store/tests/test_riak_backend.py

```diff
@@ -1034,3 +1035,37 @@ def setUp(self):
         self.state = FakeRiakState(is_sync=True)
         self.add_cleanup(self.state.teardown)
         self.set_up_tests(FakeMemoryRiakManager(self.state))
+
+
+class TestRiakBackendUtils(VumiTestCase):
+    def test_formatter_mismatched_batch_id(self):
+        """
+        When the index value passed to the formatter doesn't match what is
+        expected for the given batch ID, an error is thrown.
+        """
+        result = ("mybatch$20150730T11:03$apples", "key")
+        try:
+            key_with_ts_and_value_formatter("yourbatch", result)
+        except ValueError as e:
+            self.assertEqual(
+                str(e),
+                "Index value 'mybatch$20150730T11:03$apples' does not begin " +
+                "with expected prefix 'yourbatch$'.")
+        else:
+            self.fail()
```

Comment:
Maybe use `self.assertRaises()` here? `trial`'s implementation returns the exception it caught, so you can make further assertions on it.